### PR TITLE
feat: table filterSearch supports function

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33935
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    feat: table filterSearch supports function `(input, record) => record.value.indexOf(input) > -1`    |
| 🇨🇳 Chinese |    功能: 表格`filterSearch`属性支持方法`(input, record) => record.value.indexOf(input) > -1`    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by heiyu4585** [comment]
@MadCcc  @yoyo837 帮忙看下

**Comment by MadCcc** [comment]
代码合过了吗，这个是新开的？
如果没合过的话 ci 得看下

**Comment by heiyu4585** [comment]
> 代码合过了吗，这个是新开的？ 如果没合过的话 ci 得看下

done

File: components/table/demo/filter-search.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+## zh-CN
+
+`filterSearch` 用于开启筛选项的搜索，通过 `filterSearch:(input, record) => boolean` 设置自定义筛选方法
+
+## en-US
+
+`filterSearch` is used for making filter dropdown items searchable,use `filterSearch:(input, record) => boolean` set
+custom filter function
```
Comment (MadCcc) [inline components/table/demo/filter-search.md:16]:
```suggestion
`filterSearch` is used to enable search of filter items, and you can set a custom filter method through `filterSearch:(input, record) => boolean`.
```

Comment (heiyu4585) [reply components/table/demo/filter-search.md]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**